tmt/steps/prepare/install.py

```diff
@@ -10,6 +10,310 @@
+        self.plugin = plugin
+        self.check_sanity()
+        self.process_packages()
+        self.prepare_sudo()
+
+    def check_sanity(self):
+        """ Check parameters, bail out if nothing to do """
+        # Check rpm packages in local directories
+        self.packages = self.pkg_data.get('package', [])
+        self.directories = self.pkg_data.get('directory', [])
+        if not self.packages and not self.directories:
+            self.debug("No packages for installation found.", level=3)
+            return
+
+    def process_packages(self):
+        """ Process package names and directories """
+        # Detect local, debuginfo and repository packages
+        for package in self.packages:
+            if package.endswith('.rpm'):
+                self.local_packages.append(package)
+            elif re.search(r"-debug(info|source)(\.|$)", package):
+                # Strip the '-debuginfo' string from package name
+                # (installing with it doesn't work on RHEL7)
+                package = re.sub(r"-debuginfo((?=\.)|$)", "", package)
+                self.debuginfo_packages.append(package)
+            else:
+                self.repo_packages.append(package)
+        # Check rpm packages in local directories
+        for directory in self.directories:
+            self.info('directory', directory, 'green')
+            if not os.path.isdir(directory):
+                raise tmt.utils.PrepareError(
+                    f"Packages directory '{directory}' not found.")
+            for filename in os.listdir(directory):
+                if filename.endswith('.rpm'):
+                    self.debug(f"Found rpm '{filename}'.", level=3)
+                    self.local_packages.append(
+                        os.path.join(directory, filename))
+
+    def prepare_sudo(self):
+        """ Prepare whoami """
+        self.sudo = ""
+        self.debug('Check if sudo is necessary.', level=2)
+        user = self.guest.execute('whoami')[0].strip()
+        self.sudo = '' if user == 'root' else 'sudo '
+
+    def prepare_repository(self, **kwargs):
+        """ Configure additional repository """
+        raise NotImplementedError
+
+    def prepare_packages(self, package_list, title):
+        """ Show package info and return quoted package names """
+        # Show a brief summary by default
+        if not self.opt('verbose'):
+            summary = fmf.utils.listed(package_list, max=3)
+            self.info(title, summary, 'green')
+        # Provide a full list of packages in verbose mode
+        else:
+            summary = fmf.utils.listed(package_list, 'package')
+            self.info(title, summary + ' requested', 'green')
+            for package in sorted(package_list):
+                self.verbose(package, shift=1)
+        # Return quoted package names
+        return " ".join([tmt.utils.quote(package) for package in package_list])
+
+    def enable_copr_epel6(self, copr, guest):
```

Comment:
I suppose `guest` is not needed, `self.guest` should be used consistently by the class.